### PR TITLE
TSC/SOC compliance mapping

### PR DESCRIPTION
**Pull request by snaow**
Adding support to TSC criteria which will help all the users with the SOC compliance.
The PR includes 1030 rules and 60 SCA checks that have been mapped with TSC controls.

Related PRs
- Core/Framework: https://github.com/wazuh/wazuh/pull/4882
- Frontend Commits: https://github.com/wazuh/wazuh-kibana-app/commit/bf2d31d043307e66e4a3209b60410d800828c16b https://github.com/wazuh/wazuh-kibana-app/commit/f47e606755a8ec9446203a8e4aba09d2cca4f52c
- API: https://github.com/wazuh/wazuh-api/commit/719563a6e18581a2c062ba0f6a950730ac74222d

Example rule:

```
<rule id="3158" level="10" frequency="8" timeframe="120">
  <if_matched_sid>3108</if_matched_sid>
  <same_source_ip />
  <description>sendmail: Multiple pre-greetings rejects.</description
  <group>multiple_spam,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SI.4,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,tsc_CC7.4,</group>
</rule>
```

Source:

- [2017 Trust Services Criteria for Security, Availability, Processing Integrity, Confidentiality, and Privacy](https://www.aicpa.org/content/dam/aicpa/interestareas/frc/assuranceadvisoryservices/downloadabledocuments/trust-services-criteria.pdf)

### CODE REVIEW

**Comment by davidjiglesias** [comment]
Shouldn't ruleset VERSION be 3.13.0?

File: rules/0095-sshd_rules.xml [hunk 1]
```diff
@@ -47,21 +47,21 @@
...
-    <group>recon,pci_dss_11.4,gpg13_4.12,gdpr_IV_35.7.d,nist_800_53_SI.4,</group>
+    <group>recon,pci_dss_11.4,gpg13_4.12,gdpr_IV_35.7.d,nist_800_53_SI.4,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,tsc_CC7.4,</group>
   </rule>
 
   <rule id="5707" level="14">
     <if_sid>5700</if_sid>
     <match>fatal: buffer_get_string: bad string</match>
     <description>sshd: OpenSSH challenge-response exploit.</description>
-    <group>exploit_attempt,pci_dss_11.4,pci_dss_6.2,gpg13_4.12,gdpr_IV_35.7.d,nist_800_53_SI.4,nist_800_53_SI.2,</group>
+    <group>exploit_attempt,pci_dss_11.4,pci_dss_6.2,gpg13_4.12,gdpr_IV_35.7.d,nist_800_53_SI.4,nist_800_53_SI.2,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,tsc_CC7.4,tsc_A1.2,tsc_CC8.1,</group>
```
Comment (chemamartinez) [inline rules/0095-sshd_rules.xml:64]: A1.2 and CC8.1 don't fit this rule. The same for the rule 5714.

File: rules/0175-proftpd_rules.xml [hunk 2]
```diff
@@ -135,14 +135,14 @@
...
-    <group>service_availability,pci_dss_10.6.1,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,</group>
+    <group>service_availability,pci_dss_10.6.1,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
   </rule>
 
   <rule id="11219" level="12">
     <if_sid>11200</if_sid>
     <match>Reallocating sreaddir buffer</match>
     <description>proftpd: FTP server Buffer overflow attempt.</description>
-    <group>pci_dss_6.5.2,pci_dss_11.4,pci_dss_6.2,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,nist_800_53_SI.2,</group>
+    <group>pci_dss_6.5.2,pci_dss_11.4,pci_dss_6.2,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,nist_800_53_SI.2,tsc_CC6.6,tsc_CC7.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,tsc_CC7.4,tsc_A1.2,tsc_CC8.1,</group>
```
Comment (chemamartinez) [inline rules/0175-proftpd_rules.xml:145]: CCA1.2, CC8.1, CC6.8 and CC7.1 could not fit here

File: rules/0180-pure-ftpd_rules.xml [hunk 3]
```diff
@@ -18,21 +18,21 @@
     <if_sid>11300</if_sid>
     <match>[INFO] New connection from</match>
     <description>pure-ftpd: New FTP connection.</description>
-    <group>connection_attempt,pci_dss_10.6.1,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,</group>
+    <group>connection_attempt,pci_dss_10.6.1,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
```
Comment (chemamartinez) [inline rules/0180-pure-ftpd_rules.xml:21]: CC6.8 is related to malware detection. I think it doesn't fit in rules about connection and authentication.

File: rules/0015-ossec_rules.xml [hunk 4]
```diff
@@ -20,39 +20,39 @@
...
+    <group>pci_dss_10.6.1,gpg13_10.1,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
   </rule>
 
   <rule id="503" level="3">
     <if_sid>500</if_sid>
     <options>alert_by_email</options>
     <match>Agent started</match>
     <description>Ossec agent started.</description>
-    <group>pci_dss_10.6.1,pci_dss_10.2.6,gpg13_10.1,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,nist_800_53_AU.14,nist_800_53_AU.5,</group>
+    <group>pci_dss_10.6.1,pci_dss_10.2.6,gpg13_10.1,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,nist_800_53_AU.14,nist_800_53_AU.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,tsc_CC7.4,</group>
```
[the remark this answers is not in the document]
Comment (snaow) [reply rules/0015-ossec_rules.xml:39]: You were right, fixed.

File: rules/0020-syslog_rules.xml [hunk 5]
```diff
@@ -36,32 +36,32 @@
   <rule id="1004" level="5">
     <match>^exiting on signal</match>
     <description>Syslogd exiting (logging stopped).</description>
-    <group>pci_dss_10.6.1,gpg13_10.1,gpg13_4.14,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,</group>
+    <group>pci_dss_10.6.1,gpg13_10.1,gpg13_4.14,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_AU.6,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
```
[the remark this answers is not in the document]
Comment (snaow) [reply rules/0020-syslog_rules.xml:39]: Solved

**Pull request merged**